menas/src/main/scala/za/co/absa/enceladus/rest/services/VersionedModelService.scala

```diff
@@ -86,4 +86,11 @@ abstract class VersionedModelService[C <: VersionedModel](versionedMongoReposito
     }
   }
 
+  def validateEntityName(name: String , entityType:String ):String  = {
+    if (name.matches("""\w+""")) {
```

Comment:
My suggestion: include check for uniqueness from `create` method within this function, call `validateEntityName` within `create` and remove the call from higher instances.

Reply:
Please, if you don't agree with a comment don't just mark it as _Resolved_. I would welcome to learn why you think the suggested idea wouldn't work.

Reply:
Sorry, I meant to resolve some other conversation, I accidentally resolved this.

Reply:
> My suggestion: include check for uniqueness from `create` method within this function, call `validateEntityName` within `create` and remove the call from higher instances.

Please could you elaborate, couldnt understand.

Reply:
> 
> 
> Sorry, I meant to resolve some other conversation, I accidentally resolved this.

NP 😃

Reply:
> 
> 
> > My suggestion: include check for uniqueness from `create` method within this function, call `validateEntityName` within `create` and remove the call from higher instances.
> 
> Please could you elaborate, couldnt understand.

```
  def create(item: C, username: String): Future[C] = {
   validateEntityName(item.name)
   versionedMongoRepository.create(item, username)
   getLatestVersion(item.name)
  }

  def validateEntityName(name: String ):Boolean  = {
    if (!isUniqueName(name)) {
      throw new Exception(s"Name already exists: $name}")
    }
    if (!name.matches("""\w+""")) {
      throw new Exception(s"Entity name must not contain whitespace : $name")
    }
    true
  }
```

Reply:
We don't do it right now. That would be another/new issue.
The only place that these are checked is in JS.
But I agree that this should be implemented.

Reply:
@Zejnilovic Ok, can i go ahead and create a new issue and resolve this conversation?

Reply:
If you ask me, then yes. What has the creator @benedeki to say?
And reference the issue here.

Reply:
@Zejnilovic was not aware that the uniqueness check was already there and that I only suggested unification of the two name checks (we clarified it in conversation as I didn't understand his comment)
I don't want to stall this PR on this, this is why I made it a suggestion only, not request. I believe it make the logic less scattered and that it's not a new issue.

Reply:
Thanks Everyone, resolving this conversation